jdaviz/app.py

```diff
@@ -849,21 +849,13 @@ def remove(stack_items):
                     if viewer['id'] == cid:
                         stack['viewers'].remove(viewer)
 
-                # If the stack is empty of viewers, also delete the stack
-                if len(stack['viewers']) == 0 and \
```

Comment:
It seems like this was meant to account for groupings of viewers (two "sub viewers" in a viewer), but we don't do anything like that anywhere as far as I know. So I think it's ok to remove, considering it seems to solve the bug. 

I'll also say I'm slightly confused as to why a local function is defined here and then immediately called instead of just...doing the code that's in that function. Could have saved some indents...upon further digging it seems like it's a relic of when this worked slightly differently, see https://github.com/spacetelescope/jdaviz/commit/869617e0904a6df325ca76aac62efc7df22fabad.

Reply:
I just thought it's like that for the recursion. 🤷

Reply:
Ahh good call, I missed that there was a call to a bare `remove` in there amidst the other calls to `stack_items.remove` and 'stack['viewers'].remove'